hyperon_das/query_engines.py

```diff
@@ -255,6 +266,46 @@ def get_links(
         if not local:
             return self.remote_das.get_links(link_type, target_types, link_targets)
 
+    def get_incoming_links(
+        self, atom_handle: str, **kwargs
+    ) -> List[Union[dict, str, Tuple[dict, List[dict]]]]:
+        local_links = self.local_query_engine.get_incoming_links(atom_handle, **kwargs)
+        remote_links = self.remote_das.get_incoming_links(atom_handle, **kwargs)
+
+        if not local_links and remote_links:
+            return remote_links
+        elif local_links and not remote_links:
+            return local_links
+        elif not local_links and not remote_links:
+            return []
+
+        if kwargs.get('handles_only', False):
+            return list(set(local_links + remote_links))
+        else:
+            answer = []
+
+            if isinstance(remote_links[0], dict):
+                remote_links_dict = {link['handle']: link for link in remote_links}
+            else:
+                remote_links_dict = {
+                    link['handle']: (link, targets) for link, targets in remote_links
+                }
+
+            for local_link in local_links:
+                if isinstance(local_link, dict):
+                    handle = local_link['handle']
+                else:
+                    handle = local_link[0]['handle']
+                    local_link = local_link[0]
+
+                if handle in remote_links_dict:
+                    answer.append(local_link)
```

Comment:
Shouldn't this `append` be valled outside the `if`?

Actually, why do we need an `if` anyway? I think `dict` has a `pop` method which removes a value without throwing if you provide a default `None`. This would prevent us from making two searches in the dict.

Reply:
I'm comparing `local_links` with `remote_links`. So if `local_link` exists in `remote_links_dict` I want that local_link to be in my `answer` and that's why I `append` it to the `answer` and remove that handle from the `remote_links_dict`.

The objective is to have all `local links + remote links` in my `answer`, but If there is any similar link a get only the local_link

Reply:
I understood what you're doing. But I don't think you need the `if` in line 301 to accomplish that because I think `answer.append(local_link)` should be called anyway (I mean that current code is wrong)